bindings/go/context.go

```diff
@@ -44,8 +40,50 @@ func NewUcpContext(params *UcpParams) (*UcpContext, error) {
 	return ctx, nil
 }
 
-func (p *UcpContext) Close() error {
-	C.ucp_cleanup(p.context)
-	p.context = nil
+func (c *UcpContext) Close() error {
+	C.ucp_cleanup(c.context)
+	c.context = nil
 	return nil
 }
+
+// Mask which memory types are supported
+func (c *UcpContext) MemoryTypesMask() (uint64, error) {
+	var ucp_attrs C.ucp_context_attr_t
+
+	ucp_attrs.field_mask = C.UCP_ATTR_FIELD_MEMORY_TYPES
+
+	if status := C.ucp_context_query(c.context, &ucp_attrs); status != C.UCS_OK {
+		return 0, NewUcxError(status)
+	}
+
+	return uint64(ucp_attrs.memory_types), nil
+}
```

Comment:
maybe wrap entire ucp_context_query?
we want that language bindings will have an API similar as possible to C native API to ease documentation and usage